rqt_py_common/src/rqt_py_common/message_helpers.py

```diff
@@ -41,19 +42,123 @@
 SRV_MODE = 'srv'
 ACTION_MODE = 'action'
 
+ROSIDL_FILTERS = {
+    MSG_MODE: lambda n: n.endswith('.msg'),
+    SRV_MODE: lambda n: n.endswith('.srv'),
+    ACTION_MODE: lambda n: n.startswith('action/') and n.endswith('.idl')
+}
+
+# Currently there is no high-level action class. This namedtuple will serve as a packed
+# representation of an Action
+ActionTuple = namedtuple('Action', ['Goal', 'Result', 'Feedback'])
+
+
+def _strip_type_name(type_name):
+    prefixes = ['msg/', 'srv/', 'action/']
+    suffixes = ['.msg', '.srv', '.idl']
+    for prefix in prefixes:
+        if type_name.startswith(prefix):
+            type_name = type_name[len(prefix):]
+            break
+    for suffix in suffixes:
```

Comment:
I'm not a fan of the two for loops. Why not use `zip`?

Reply:
I suppose it's a little less likely to be buggy with zip.